File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public static Boolean isPalindrome(String s,int i, int j){
        while(i<=j){
            if(s.charAt(i)!=s.charAt(j)) return false;
            i++;
            j--;
        }
        return true;
    }
    public String longestPalindrome(String s) {
        int maxi = 0;
        int n = s.length();
        if(n==1) return s;
        String ans = "" + s.charAt(0);
        for(int i=0;i<n;i++){
            for(int j=i;j<n;j++){
                if(isPalindrome(s,i,j) && maxi<(j-i+1)){
                    maxi = j-i+1;
                    ans = s.substring(i,j+1);
                }
            }
        }
        return ans;
    }
}

File: 5-longest-palindromic-substring/longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def isPalindrome(s):
            return s == s[::-1]
        ans = ""
        maxi = 0
        n = len(s)
        for i in range(n):
            for j in range(i+1,n+1):
                if isPalindrome(s[i:j]) and maxi<(j-i+1):
                    maxi = j-i+1
                    ans = s[i:j]
        return ans